temporal-sdk/src/main/java/io/temporal/internal/statemachines/WorkflowStateMachines.java

```diff
@@ -148,11 +151,25 @@ enum HandleEventStatus {
 
   private final WFTBuffer wftBuffer = new WFTBuffer();
 
-  /** */
   private List<Message> messages = new ArrayList<Message>();
 
-  public WorkflowStateMachines(StatesMachinesCallback callbacks) {
-    this(callbacks, (stateMachine) -> {});
+  private final SdkFlags flags;
+
+  public WorkflowStateMachines(
+      StatesMachinesCallback callbacks, GetSystemInfoResponse.Capabilities capabilities) {
+    this(callbacks, (stateMachine) -> {}, capabilities);
+  }
+
+  @VisibleForTesting
+  public WorkflowStateMachines(
+      StatesMachinesCallback callbacks,
+      Functions.Proc1<StateMachine> stateMachineSink,
+      GetSystemInfoResponse.Capabilities capabilities) {
+    this.callbacks = Objects.requireNonNull(callbacks);
+    this.commandSink = cancellableCommands::add;
+    this.stateMachineSink = stateMachineSink;
+    this.localActivityRequestSink = (request) -> localActivityRequests.add(request);
```

Comment:
```suggestion
    this.localActivityRequestSink = localActivityRequests::add;
```

(didn't check this)

Reply:
kept the code as is because this does not work